src/actions/generalActions.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Callback function invoked by the dispatcher upon any defined actions that get raised by the View 
+ * (SVG overlay or control buttons).
+ * A switch statement is used to handle most non-tool related actions, while tool related actions are 
+ * delegated to ToolHandler. 
+ * @param {*} model 
+ * @param {*} toolHandler 
+ */
+const reactToGeneralAction = (model, toolHandler) => {
+	return (action) => {
+		switch (action.type) {
+      /**
+       * Action to switch tools, or reset the currently active tool. Achieved by clicking tool buttons 
+       * ont the toolbar, or by using any defined hotkeys.
+       */
+			case 'MODE_UPDATE':
+        /**
+         * If mode change occurs during an annotation activity, due to erraneous click or hotkey press
+         * Clean up annotation being created
+         */
+				if (model.activityInProgress === true) {
+          // Reset click, conduct tool specific cleanup
+          toolHandler.cleanTool();
+          model.activityInProgress = false;
+        }
+
+        /**
+         * action.mode defines the payload sent to the dispatcher. Defines the mode to be updated to. Avoid
+         * re-initialising if current tool is pressed for an nth time. 
+         */
+        if (model.mode !== action.mode) {
```

Comment:
Removing stuff we dont need will definitely clear up. 
It seems like two things must be checked on every nav event - activity in progress and model.mode.  What are possible values of model.mode?

Can this check for both variables be uniformly included in every switch statement. Sometimes we check current value of activityInProgress before flipping it, sometimes we dont. Inconsistencies like that bug me.